dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/lifecycle/BeanLifecycleTest.java

```diff
@@ -0,0 +1,132 @@
+                            "name": "sayHello",
+                            "arguments": {
+                              "name": "World"
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, "/beanLifecycleTest", request);
+
+        // Strict Mode tests
+        String expectedResponseString = """
+                        {
+                          "id": "2",
+                          "jsonrpc": "2.0",
+                          "result": {
+                            "content": [
+                              { "type": "text", "text": "Hello, World" }
+                            ],
+                            "isError": false
+                          }
+                        }
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+
+        // Retrieve lifecycle logs since the mark
+        List<String> logs = server.findStringsInLogsAndTraceUsingMark("[LIFECYCLE]");
+
+        // Quick sanity checks on log presence of LIFECYCEL
+        assertTrue("Expected PostConstruct lifecycle log", logs.stream().anyMatch(line -> line.contains("[LIFECYCLE]")));
+        server.waitForStringInLog("\\[LIFECYCLE] @PostConstruct ClassTool", 10_000);
+
+        assertTrue("Expected PreDestroy lifecycle log", logs.stream().anyMatch(line -> line.contains("[LIFECYCLE]")));
+        server.waitForStringInLog("\\[LIFECYCLE] @PreDestroy ClassTool", 10_000);
+
+        // Retrieve full list of lifecylce log
+        java.util.List<String> lifecycle = server.findStringsInLogsAndTraceUsingMark(".*\\[LIFECYCLE].*");
+
+        assertFalse("No [LIFECYCLE] lines found since mark", lifecycle.isEmpty());
+
+        boolean hasPost = lifecycle.stream().anyMatch(l -> l.contains("@PostConstruct ClassTool"));
+        boolean hasPre = lifecycle.stream().anyMatch(l -> l.contains("@PreDestroy ClassTool"));
+
+        // If PreDestroy was not logged, fail the test
+        if (!hasPre) {
+            fail("Did not find @PreDestroy line. Lifecycle lines since mark:\n" + String.join("\n", lifecycle));
+        }
+        assertTrue("Missing @PostConstruct", hasPost);
+
+        int postIdx = indexOfFirst(lifecycle, "@PostConstruct ClassTool");
+        int preIdx = indexOfFirst(lifecycle, "@PreDestroy ClassTool");
+        assertTrue("Expected @PostConstruct before @PreDestroy.\n" + String.join("\n", lifecycle),
+                   postIdx >= 0 && preIdx >= 0 && postIdx < preIdx);
+    }
+
+    // Returns the index of the first log line containing the specified search term
+    private static int indexOfFirst(java.util.List<String> lines, String needle) {
+        for (int i = 0; i < lines.size(); i++) {
+            if (lines.get(i).contains(needle))
+                return i;
+        }
+        return -1;
+    }
```

Comment:
There are a couple of things that need fixing, and a couple that we can tidy up.

To fix:
- first wait for `@PreDestroy` message, then extract all `[LIFECYCLE]` messages
- don't provide a manual timeout value for the wait
- don't search trace for messages, otherwise you'll find all the messages twice

To tidy up:
- when you've extracted all the log messages you can do this to check the right messages are emitted in order
  ```java
  assertThat(lifecycleMessages, contains(contatinsString("@PostConstruct ClassTool"), containsString("@PreDestroy ClassTool")));
  ```